Tests/DeepLearningTests/TensorTests.swift

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import DeepLearning
+
+final class TensorTests: XCTestCase {
+    func testSimpleCond() {
+        func selectValue(_ pred: Bool) -> Tensor<Int32> {
+            let a = Tensor<Int32>(0)
+            let b = Tensor<Int32>(1)
+            if pred {
+                return a
+            }
+            return b
+        }
+
+        XCTAssertEqual(0, selectValue(true).scalar)
+    }
+
+    @inline(never)
+    func testRankGetter() {
+        let tensor = Tensor<Int32>(shape: [3, 4, 5], scalars: Array(0..<60))
+        XCTAssertEqual(3, tensor.rank)
+    }
+
+    // TODO: Merge all rank/shape getter tests into one when we support code motion to avoid sends.
+
+    @inline(never)
+    func testRankGetter2() {
+        let vector = Tensor<Int32>([1])
+        XCTAssertEqual(1, vector.rank)
+    }
+
+    @inline(never)
+    func testRankGetter3() {
+        let matrix = Tensor<Float>([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]])
+        XCTAssertEqual(2, matrix.rank)
+    }
+
+    @inline(never)
+    func testRankGetter4() {
+        let ones = Tensor<Int32>(ones: [1, 2, 2, 2, 2, 2, 1])
+        XCTAssertEqual(7, ones.rank)
+    }
+
+    @inline(never)
+    func testShapeGetter() {
+        let tensor = Tensor<Int32>(shape: [3, 4, 5], scalars: Array(0..<60))
+        XCTAssertEqual([3, 4, 5], tensor.shape)
+    }
+
+    @inline(never)
+    func testShapeGetter2() {
+        let vector = Tensor<Int32>([1])
+        XCTAssertEqual([1], vector.shape)
+    }
+
+    @inline(never)
+    func testShapeGetter3() {
+        let matrix = Tensor<Float>([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]])
+        XCTAssertEqual([2, 3], matrix.shape)
+    }
+
+    @inline(never)
+    func testShapeGetter4() {
+        let ones = Tensor<Int32>(ones: [1, 2, 2, 2, 2, 2, 1])
+        XCTAssertEqual([1, 2, 2, 2, 2, 2, 1], ones.shape)
+    }
+}
```

Comment:
In order to make tests run on Linux, could you add a `allTests` computed property like [this one](https://github.com/tensorflow/swift-apis/blob/3f716844e38b462cf6e6c0ba70b56b67fa5cf9fb/Tests/DeepLearningTests/LayerTests.swift#L123) in every test file?